pkg/controller/template/template_controller.go

```diff
@@ -403,9 +404,10 @@ func (ctrl *Controller) syncControllerConfig(key string) error {
 		if err := ctrl.syncRunningStatus(cfg); err != nil {
 			return err
 		}
-	} else {
-		glog.V(2).Info("ControllerConfig didn't change, skipping templates sync")
-		return nil
+	}
+
+	if cfg.Spec.Version != version.Hash {
+		return ctrl.syncFailingStatus(cfg, fmt.Errorf("refusing to retemplate with different ControllerConfig and MCC versions: %s != %s", cfg.Spec.Version, version.Hash))
```

Comment:
OK, so if the MCO has updated but the MCC hasn't, the MCC will block until it's updated.  Makes sense.

Reply:
this should guard also the case where, somehow, the new MCC is running with new templates but the controllerconfig is the old one and viceversa